python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py

```diff
@@ -25,39 +28,70 @@ def __init__(self, context):  # pylint: disable=too-many-locals
         # Extract parameters from resource config. Note that we can't pass None values to
         # snowflake.connector.connect() because they will override the default values set within the
         # connector; remove them from the conn_args dict.
-        self.conn_args = {
-            k: context.resource_config.get(k)
-            for k in (
-                "account",
-                "user",
-                "password",
-                "database",
-                "schema",
-                "role",
-                "warehouse",
-                "autocommit",
-                "client_prefetch_threads",
-                "client_session_keep_alive",
-                "login_timeout",
-                "network_timeout",
-                "ocsp_response_cache_filename",
-                "validate_default_parameters",
-                "paramstyle",
-                "timezone",
-            )
-            if context.resource_config.get(k) is not None
-        }
+        self.connector = context.resource_config.get("connector", None)
+
+        if self.connector == 'sqlalchemy':
+            self.conn_args = {
+                k: context.resource_config.get(k)
+                for k in (
+                    "account",
+                    "user",
+                    "password",
+                    "database",
+                    "schema",
+                    "role",
+                    "warehouse",
+                    "cache_column_metadata",
+                    "numpy",
+                )
+                if context.resource_config.get(k) is not None
+            }
+
+        else:
+            self.conn_args = {
+                k: context.resource_config.get(k)
+                for k in (
+                    "account",
+                    "user",
+                    "password",
+                    "database",
+                    "schema",
+                    "role",
+                    "warehouse",
+                    "autocommit",
+                    "client_prefetch_threads",
+                    "client_session_keep_alive",
+                    "login_timeout",
+                    "network_timeout",
+                    "ocsp_response_cache_filename",
+                    "validate_default_parameters",
+                    "paramstyle",
+                    "timezone",
+                )
+                if context.resource_config.get(k) is not None
+            }
+
 
         self.autocommit = self.conn_args.get("autocommit", False)
         self.log = context.log_manager
 
     @contextmanager
-    def get_connection(self):
-        conn = snowflake.connector.connect(**self.conn_args)
-        yield conn
-        if not self.autocommit:
-            conn.commit()
-        conn.close()
+    def get_connection(self, raw_conn=True):
```

Comment:
Any particular reason to default to the raw connection?  I've used the non-raw connection in the past, but am not super familiar with the distinction.

Another thought - for a user who isn't using sqlalchemy, what would be the meaning of the "raw_conn" argument?

Reply:
I mention this at the end of the main PR comment: 
```
Also note, the added parameter raw_conn=False is needed with get_connection because the .to_sql function won't accept a raw connection from sqlalchemy. However, Dagster's execute_query function require the raw_connection to maintain existing default behavior.
```
Basically I default to the raw connection because this is more consistent with the default behavior.  But to use the pandas `.to_sql` function I alternatively need the ability to use the non-raw connection as well (to be specific, I believe the existing `execute_query` function will break with the non-raw connection, but works with the raw connection.

As far as users who aren't using sqlalchemy, the `raw_conn` param would not be meaningful for them.